src/scenes/ProgressReports/EditForm/Steps/step.types.ts

```diff
@@ -7,10 +7,28 @@ export type StepComponent = ComponentType<ReportProp> & {
   isIncomplete?: IsIncompleteFn;
 };
 
+export enum NotificationSeverity {
+  Required = 'required',
+  Suggested = 'suggested',
+}
+export type NotificationSeverityType = `${NotificationSeverity}`;
+
 export type IsIncompleteFn = (
   report: ReportProp & { currentUserRoles: ReadonlySet<Role> }
-) => boolean;
+) => {
+  isIncomplete: boolean;
+  severity: NotificationSeverityType;
+};
 
 export type GroupedStepMapShape = {
   [Section in string]: ReadonlyArray<[label: string, component: StepComponent]>;
 };
+
+export type GroupedStepMapShapeWithSeverity = {
+  [Section in keyof GroupedStepMapShape]: Array<
+    [
+      ...GroupedStepMapShape[Section][number],
+      severity: NotificationSeverityType
```

Comment:
I wouldn't bother trying to spread from the other type here. While clever, it's confusing. And this type also has nothing to do with the other one.
```suggestion
      label: string,
      component: StepComponent,
      severity: NotificationSeverityType
```

Alternatively, you could use an object instead of a tuple here. That might be clearer everywhere.
```ts
ReadonlyArray<{
  label: string;
  step: StepComponent;
  severity: IncompleteSeverity;
}>